onnxruntime/core/providers/dnnl/subgraph/dnnl_qattention.cc

```diff
@@ -189,19 +188,16 @@ void DnnlQAttention::CreatePrimitive(DnnlSubgraphPrimitive& sp, DnnlNode& node)
     auto total_scale_mem = ComputeTotalScale(sp, node);
 
     auto bias_md = sp.GetMemory(node.Input(BIAS)).get_desc();
-    bias_md = bias_md.reshape({1, 1, bias_md.dims()[0]});
+    bias_md = bias_md.reshape({1, 1, bias_md.get_dims()[0]});
     auto QKV_desc = QKV_mem.get_desc();
 
-    //always broadcast from bias to QKV
-    auto binary_d = dnnl::binary::desc(dnnl::algorithm::binary_add, QKV_desc, bias_md, QKV_desc);
-
     dnnl::primitive_attr binary_attr;
     //scale source 0, matmul output
     if (total_scale_mem) {
-      binary_attr.set_scales(DNNL_ARG_SRC_0, 0, {DNNL_RUNTIME_F32_VAL});
+      binary_attr.set_scales_mask(DNNL_ARG_SRC_0, 0);
     }
 
-    auto binary_pd = dnnl::binary::primitive_desc(binary_d, binary_attr, eng);
+    auto binary_pd = dnnl::binary::primitive_desc(eng, dnnl::algorithm::binary_add, QKV_desc, bias_md, QKV_desc, binary_attr);
```

Comment:
Something here looks/feel wrong but I can not pin point why. Can you double check this is correct? Sorry that I can not give an exact thing to check.